src/megaapi_impl.cpp

```diff
@@ -20674,27 +20979,181 @@ void MegaApiImpl::sendPendingRequests()
                 break;
             }
 
-            MegaSyncPrivate *sync = new MegaSyncPrivate(localPath, node->nodehandle, -nextTag);
+            auto nextSyncTag = client->nextSyncTag();
+            MegaSyncPrivate *sync = new MegaSyncPrivate(localPath, node->nodehandle, nextSyncTag);
             sync->setListener(request->getSyncListener());
             sync->setRegExp(request->getRegExp());
 
-            SyncConfig syncConfig{localPath, request->getNodeHandle(), 
+            SyncConfig syncConfig{nextSyncTag, localPath, request->getNodeHandle(),
                                   static_cast<fsfp_t>(request->getNumber()), 
                                   regExpToVector(request->getRegExp())};
-            e = client->addsync(std::move(syncConfig), DEBRISFOLDER, NULL, -nextTag, sync);
+
+            syncerror_t syncError = NO_SYNC_ERROR;
+            e = client->addsync(syncConfig, DEBRISFOLDER, NULL, syncError, true, sync);
+            request->setNumDetails(syncError);
             if (!e)
             {
                 Sync *s = client->syncs.back();
                 fsfp_t fsfp = s->fsfp;
+                sync->setError(syncError);
                 sync->setState(s->state);
                 sync->setLocalFingerprint(fsfp);
+                sync->setMegaFolderYielding(getNodePathByNodeHandle(node->nodehandle));
                 request->setNumber(fsfp);
-                syncMap[-nextTag] = sync;
+                request->setTransferTag(nextSyncTag);
+                syncMap[nextSyncTag] = sync;
+
+                fireOnSyncAdded(sync, MegaSync::NEW);
+                fireOnRequestFinish(request, MegaError(API_OK));
+                break;
+            }
+
+            delete sync;
+            break;
+        }
+        case MegaRequest::TYPE_ENABLE_SYNC:
+        {
+            auto tag = request->getNumDetails();
+            auto sync_it = syncMap.find(tag);
+            if ( sync_it == syncMap.end() || !sync_it->second)
+            {
+                e = API_ENOENT;
+                break;
+            }
+
+            syncerror_t syncError = NO_SYNC_ERROR;
+
+            e = client->enableSync(tag, syncError, true);
+            //TODO: note to reviewer: in case this fails with a non temporary error, this sets the sync
+            // to FAILED (which oddly entails: isEnabled = true: from the user perspective it has been enabled)
+            // Alternatively, we could keep/enforce a SYNC_DISABLED+NO_SYNC_ERROR (i.e. manually disabled) and
+            //  asume the failure in the req finish is enough for the app and that we do not want to update the sync configuration
```

Comment:
So, I'm leaving it as is. The enabling request will return an error if the sync couldn't be activated. Be it because of a temporary or non-temporary error.

Reply:
comment updated here anyway: https://github.com/meganz/sdk/pull/2091